From cbbf1d573baa35fe20c50d48d6498944dad149fe Mon Sep 17 00:00:00 2001
From: Aparna Mallavarapu <aparnam@codeaurora.org>
Date: Wed, 28 Nov 2012 13:14:56 +0530
Subject: [PATCH 57/75] arm: board-msm7627a: Add TS support for 8x25Q EVBD

Add TS support for 8x25Q-EVB since it uses a different
irq_gpio.

Change-Id: Ifb4d42d0ab7c5da3f4a41996fa60375eeab2a4fa
Signed-off-by: Aparna Mallavarapu <aparnam@codeaurora.org>
---
 arch/arm/mach-msm/board-msm7627a-io.c |   13 ++++++++++++-
 1 files changed, 12 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-msm/board-msm7627a-io.c b/arch/arm/mach-msm/board-msm7627a-io.c
index 5ba51fc..52375f5 100644
--- a/arch/arm/mach-msm/board-msm7627a-io.c
+++ b/arch/arm/mach-msm/board-msm7627a-io.c
@@ -608,6 +608,7 @@ static struct platform_device hs_pdev = {
 #define FT5X06_RESET_GPIO	26
 
 #define FT5X16_IRQ_GPIO		122
+#define FT5X16_IRQ_GPIO_8625Q	115
 
 static ssize_t
 ft5x06_virtual_keys_register(struct kobject *kobj,
@@ -673,7 +674,7 @@ static void __init ft5x06_touchpad_setup(void)
 	int rc;
 	int irq_gpio;
 
-	if (machine_is_qrd_skud_prime() || machine_is_msm8625q_evbd()) {
+	if (machine_is_qrd_skud_prime()) {
 		irq_gpio = FT5X16_IRQ_GPIO;
 
 		ft5x06_platformdata.x_max = 540;
@@ -683,6 +684,16 @@ static void __init ft5x06_touchpad_setup(void)
 		ft5x06_device_info[0].irq = MSM_GPIO_TO_INT(FT5X16_IRQ_GPIO);
 
 		ft5x06_virtual_keys_attr.show = &ft5x16_virtual_keys_register;
+	} else if(machine_is_msm8625q_evbd()) {
+		irq_gpio = FT5X16_IRQ_GPIO_8625Q;
+
+		ft5x06_platformdata.x_max = 540;
+		ft5x06_platformdata.y_max = 960;
+		ft5x06_platformdata.irq_gpio = FT5X16_IRQ_GPIO_8625Q;
+
+		ft5x06_device_info[0].irq = MSM_GPIO_TO_INT(FT5X16_IRQ_GPIO_8625Q);
+
+		ft5x06_virtual_keys_attr.show = &ft5x16_virtual_keys_register;
 	} else
 		irq_gpio = FT5X06_IRQ_GPIO;
 
-- 
1.7.0.4

